fractal_server/app/routes/api/v2/task_collection.py

```diff
@@ -242,6 +253,85 @@ async def collect_tasks_pip(
     return state
 
 
+@router.post("/collect/custom/", status_code=201)
+async def collect_task_custom(
+    task_collect: TaskCollectCustomV2,
+    user: User = Depends(current_active_verified_user),
+    db: DBSyncSession = Depends(get_sync_db),
+):
+
+    settings = Inject(get_settings)
+
+    if task_collect.package_root is None:
+        if task_collect.package_name is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=(
+                    "Must provide at least one of 'package_root' "
+                    "and 'package_name'."
+                ),
+            )
+        if settings.FRACTAL_RUNNER_BACKEND == "slurm_ssh":
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Cannot infer 'package_root' with 'slurm_ssh' backend.",
+            )
+        res = subprocess.run(  # nosec
```

Comment:
We are lacking a bit of exception handling here:
1. If `res.returncode != 0`, we should return an informative 422 response
2. ~~If `next` hits a StopIteration, we should return an informative 422 response~~ I see this is handled with the `None` default